packages/fs-detectors/src/detect-framework.ts

```diff
@@ -135,6 +135,31 @@ async function matches(
   };
 }
 
+function removeSupercededFramework(
+  matches: (Pick<Framework, 'supercedes' | 'slug'> | null)[],
+  slug: string
+) {
+  const index = matches.findIndex(f => f?.slug === slug);
+  if (index !== -1) {
+    const framework = matches[index]!;
```

Comment:
If you hoist this out of the `if`, you might be able to remove the `!` by checking for truthyness in the `if` statement